### PR TITLE
Readme

### PR DESCRIPTION
**Pull request by wblankenship**
Updates to readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,28 @@
-# docker-node
+# NodeSource Docker Images
 
-Dockerfiles for building docker images with the [NodeSource Node.js](https://github.com/nodesource/distributions) binaries baked in.
+The NodeSource docker images deliver [NodeSource's deb and rpm](deb.nodesource.com) packages across all of our supported platforms! We offer version pinning, allowing your project to track major, minor, or patch versions of Node or iojs.
 
-# Structure
+# Usage
+
+Use any one of our images as a base for your image. We suggest caching your `package.json` and `npm install` in layers to reduce build time:
```
Comment (phillipj) [inline README.md]: Really? A couple of very wise men (@brycebaril & @thlorenz) once said the best approach was to get important dependency patches automatically. As @thlorenz discussed in a blog post about [Building Efficient Dockerfiles](http://bitjudo.com/blog/2014/03/13/building-efficient-dockerfiles-node-dot-js/).

Have you stopped advocating automatic dependency patches?

Comment (wblankenship) [inline README.md]: Hey @phillipj!

The conversation you linked to happened before my time, first time reading it!

So here are my thoughts, they may not necessarily jive with that of my co-workers, company, or best practices. Like all thoughts, these are a work in process that I'm constantly iterating on so I enjoy conversations like these!

The problem we are seeing here with dependencies and automatic patching isn't limited to just npm. Every package manager suffers from this, as does every network dependency inside of a Dockerfile. The possibility for the network dependency to change between builds inherently makes your build non-repeatable.

We accept this risk in favor of the alternative: bundling all dependencies inside of the build context prior to running `docker build`. If you are genuinely concerned about grabbing the latest version of network dependencies at build time, docker ships with the `--no-cache`  ensuring that your build grabs the latest version.

The ability for images to fall out of sync extends beyond the Dockerfile when we are talking about network dependencies as well. If Tom builds his docker image on Tuesday and Fred on Wednesday, but the network dependency changes overnight, then Fred will still have a different local image. The only way around this is rebuilding before every run. This is compounded by upstream images (like `buildpack-deps`) receiving updates that you don't pull into your local cache, so you would need to run `docker pull` before every build as well.

Long story short: If you are using docker images as artifacts that are in a certified "known state", you should be shipping the output of `docker save` and not your `Dockerfile`. If you are treating your docker images as cookbooks for spinning up an environment, and it is okay for things to break once in a blue moon because an upstream module patch turned out to be a breaking change, you can safely build from a Dockerfile.

Comment (phillipj) [inline README.md]: Thanks for sharing your thoughts @wblankenship!

You have a good point about thinking about *what* we're actually using these docker images for. There's probably not one solution which covers all scenarios. On my way to production I would like to know that I run the same combination of deps as I ran in CI before deploying anyways.

Comment (wblankenship) [inline README.md]: Thanks @phillipj :smile:

I am playing around with some concepts for Docker and CI for that very purpose. I created a tool called Dante which builds an image, then runs your CI tests on top of that image before certifying it production ready.

https://github.com/wblankenship/dante

Its still a work in progress, and more of a thought experiment than an actual tool. But interested in feedback!
